apollo-runtime/src/main/java/com/apollographql/apollo/internal/RealApolloCall.java

```diff
@@ -375,7 +380,18 @@ private ApolloInterceptorChain prepareInterceptorChain(Operation operation) {
     HttpCachePolicy.Policy httpCachePolicy = operation instanceof Query ? this.httpCachePolicy : null;
     ResponseFieldMapper responseFieldMapper = responseFieldMapperFactory.create(operation);
 
-    List<ApolloInterceptor> interceptors = new ArrayList<>(applicationInterceptors);
+    List<ApolloInterceptor> interceptors = new ArrayList<>();
+
+    if (!applicationInterceptorFactories.isEmpty() && !applicationInterceptors.isEmpty()) {
+      throw new IllegalArgumentException("You can either use applicationInterceptors or applicationInterceptorFactories "
```

Comment:
I wonder if this check is something we should do earlier in `ApolloClient.Builder.build()` method.

Reply:
Yep, definitely. I put it in the `ApolloClient` constructor. I don't think anyone uses that directly but better safe than sorry.